edsnlp/pipelines/core/sentences/sentences.pyx

```diff
@@ -90,16 +116,21 @@ cdef class SentenceSegmenter(object):
             is_newline = Lexeme.c_check_flag(token.lex, IS_SPACE) and token.lex.orth == self.newline_hash
 
             if seen_period or seen_newline:
-                if seen_period and Lexeme.c_check_flag(token.lex, IS_DIGIT):
-                    continue
                 if is_in_punct_chars or is_newline or Lexeme.c_check_flag(token.lex, IS_PUNCT):
                     continue
+                if seen_period and Lexeme.c_check_flag(token.lex, IS_DIGIT):
+                    continue
+                    seen_newline = False
+                    seen_period = False
```

Comment:
This is a correction. Before, a text like "Mesure du H.3 négatif" produced two sentences, "Mesure du H.3" and "négatif" since `seen_newlines` and `seen_period` were not reset.